sdk/python/kfp/components/component_factory_test.py

```diff
@@ -45,6 +47,15 @@ def test_with_packages_to_install_with_pip_index_url(self):
         for package in packages_to_install + pip_index_urls:
             self.assertTrue(package in concat_command)
 
+    def test_output_named_Output(self):
+
+        with self.assertRaisesRegex(ValueError,
+                                    r'"Output" is an invalid parameter name.'):
+
+            @component
+            def comp(Output: OutputPath(str), text: str):
```

Comment:
The test case you included is definitely the more critical one, but WDYT about also adding a test that has a `-> str` return annotation alongside an `Output: OutputPath` annotation? This helps ensure that the exception is thrown in both cases (yes name collision and no name collision).